service/PayTrace/PayTracePaymentServices.xml

```diff
@@ -0,0 +1,493 @@
+                </if>
+                <!-- if no gatewayCimId, store the PaymentMethod on CIM -->
+                <if condition="!paymentMethod.gatewayCimId">
+                    <service-call name="PayTrace.PayTraceServices.store#CustomerPaymentMethod"
+                            in-map="[paymentMethodId:paymentMethod.paymentMethodId, paymentId:paymentId, paymentGatewayConfigId:paymentGatewayConfigId]"/>
+                    <set field="paymentMethod" from="payment.'mantle.account.method.PaymentMethod'"/>
+                </if>
+                <!-- get party after CIM check so that gatewayCimId will be populated -->
+                <set field="party" from="paymentMethod.'Owner#mantle.party.Party'"/>
+                <if condition="!party.gatewayCimId"><return message="Party ${party.partyId} has no gateway CIM ID, not processing payment"/></if>
+                <if condition="!paymentMethod.gatewayCimId"><return message="Payment Method ${paymentMethod.paymentMethodId} has no gateway CIM ID, not processing payment"/></if>
+ 
+                <entity-find-one entity-name="PayTrace.PaymentGatewayPayTrace" value-field="pgpt"/> 
+                <if condition="pgpt == null"><return error="true" message="No PaymentGatewayPayTrace found with ID ${paymentGatewayConfigId}"/></if>
+
+                <set field="paymentRefNum" from="payment.paymentRefNum"/>
+                <if condition="!paymentRefNum">
+                     <service-call name="mantle.account.PaymentServices.get#AuthorizePaymentGatewayResponse" in-map="context" out-map="context"/>
+                     <if condition="!paymentGatewayResponse"><return error="true" message="Could not find successful authorization for Payment ${paymentId}"/></if>
+                    <set field="paymentRefNum" from="paymentGatewayResponse.referenceNum"/>
+                </if>
+                <if condition="!paymentRefNum"><return error="true" message="Could not find authorization transaction ID (reference number) for Payment ${paymentId}"/></if>
+                
+                <!-- assemble the full request Map -->
+                <set field="requestMap" from="[transaction_id:paymentRefNum]"/>
+
+                <service-call name="PayTrace.PayTracePaymentServices.payTrace#login" in-map="context" out-map="token"/>
+
+                <set field="paymentOperationEnumId" value="PgoRelease"/>
+
+                <script><![CDATA[
+                    org.moqui.util.RestClient restClient = ec.service.rest().method(org.moqui.util.RestClient.POST)
+                        .addHeader("Authorization", "bearer ${token.token.access_token}")
+                        .addHeader("Content-Type", "application/json").jsonObject(requestMap)
+                    restClient.uri().protocol("https").host(pgpt.transactionUrl).port(443).path(pgpt.apiVersion).path("transactions").path("authorization").path("capture").build()
```

Comment:
this is performing another "capture", which is incorrect. I think the correct endpoint is "void"
https://developers.paytrace.com/support/home#14000044724